static/js/activity.js

```diff
@@ -18,10 +18,16 @@ exports.IDLE = "idle";
 // case after a server-initiated reload.
 exports.has_focus = document.hasFocus && document.hasFocus();
 
+// We initialize this to false as when opened in browser this should remain
+// false. It is updated true only by the desktop app using electron_bridge
+// and hence would always remain false for all other platforms.
+exports.desktop_has_focus = false;
```

Comment:
Is the idea that code in the desktop app sets this back to false periodically when the app thinks the user is idle?

Reply:
Yes correct. When the webapp believes the user is idle and requests the desktop app to check the same, and if the the desktop app finds that the app is not in focus then sets`desktop_has_focus` back to false.